src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs

```diff
@@ -0,0 +1,1497 @@
+                OnPropertyChanged(nameof(LaunchPage));
+                OnPropertyChanged(nameof(HasLaunchOption));
+                OnPropertyChanged(nameof(WorkingDirectory));
+                OnPropertyChanged(nameof(SelectedCommandName));
+
+                OnPropertyChanged(nameof(SSLEnabled));
+                OnPropertyChanged(nameof(SSLUrl));
+                OnPropertyChanged(nameof(ApplicationUrl));
+                OnPropertyChanged(nameof(WindowsAuthenticationEnabled));
+                OnPropertyChanged(nameof(AnonymousAuthenticationEnabled));
+
+
+                SetLaunchType();
+
+                OnPropertyChanged(nameof(IsCustomType));
+                OnPropertyChanged(nameof(IsCommand));
+                OnPropertyChanged(nameof(IsExecutable));
+                OnPropertyChanged(nameof(IsProject));
+                OnPropertyChanged(nameof(IsProfileSelected));
+                OnPropertyChanged(nameof(DeleteProfileEnabled));
+                OnPropertyChanged(nameof(ShowVersionSelector));
+
+                if (oldProfile != null && !UseSpecificRuntime)
+                {
+                    oldProfile.SDKVersion = null;
+                }
+
+                UseSpecificRuntime = true;
+                
+                SetEnvironmentGrid(oldProfile);
+
+                UpdateActiveProfile();
+            }
+            finally
+            {
+                PopIgnoreEvents();
+            }
+        }
+
+        private void UpdateActiveProfile()
+        {
+            // need to set it dirty so Apply() actually saves the profile
+            // Billhie: this causes hangs. Disabling for now
+            //if (this.ParentControl != null)
+            //{
+            //    this.ParentControl.IsDirty = true;
+            //    WaitForAsync<int>(this.ParentControl.Apply);
+            //}
+        }
+
+        /// <summary>
+        /// Functions which actually does the save of the settings. Persists the changes to the launch settings
+        /// file and configures IIS if needed.
+        /// </summary>
+        public async virtual System.Threading.Tasks.Task SaveLaunchSettings()
+        {
+            ILaunchSettingsProvider provider = GetDebugProfileProvider();
+            if (EnvironmentVariables != null && EnvironmentVariables.Count > 0)
+            {
+                SelectedDebugProfile.MutableEnvironmentVariables = EnvironmentVariables.CreateDictionary();
+            }
+            else if (SelectedDebugProfile != null)
+            {
+                SelectedDebugProfile.MutableEnvironmentVariables = null;
+            }
+            
+            //await provider.UpdateAndSaveSettingsAsync(new LaunchSettings(DebugProfiles, GetIISSettings(), SelectedDebugProfile != null ? SelectedDebugProfile.Name : null));
+            await provider.UpdateAndSaveSettingsAsync(new LaunchSettings(DebugProfiles, null, SelectedDebugProfile != null ? SelectedDebugProfile.Name : null)).ConfigureAwait(false);
```

Comment:
>null [](start = 88, length = 4)

Note that this will wipe our IIS settings from the. I would instead pass back in the global settings